### ISSUE TITLE
project proposal

### ISSUE THREAD

**Issue by Nikoleta-v3** [Nikoleta-v3/Inferring-strategies-in-repeated-games#1]
Inferring strategies in repeated games: The French Defence
===========================================================

Background
----------

In a repeated game, players interact over a finite or infinite
number of turns with the aim of maximizing their payoff. Ideally, a player would
want to choose a best response to their co-player's strategy, which is the
strategy or strategies that produce the most favourable outcome for the player,
given the co-player's strategies.

However, players do not have perfect information about their co-player's
strategy, and they need to accumulate information to infer their co-player's
strategy. Bayesian inference has been previously used for this purpose [1].

To play a best response, a player needs to infer their co-player's strategy, but
in a finite game, the player has a limited number of turns to gather
information. This leads to the question of how many turns a player should spend
on exploration (trying to understand the co-player's strategy) and how many
turns on exploitation (playing the best response against the co-player).

Question
--------

We want to explore the opening sequences of a co-player and determine
the most effective sequence for accurately inferring their strategy.
Additionally, we aim to identify the minimum sequence size required to achieve
this outcome.

Methodology
------------

We'll assume the co-player's strategy set consists of pure
memory-one strategies and develop a Bayesian inference algorithm to identify the
co-player's strategy from one of the sixteen pure strategies. We'll run a
program that tests different combinations of opening moves (up to a maximum size
of n) to infer the co-player. We'll determine the sequence that has the highest
success rate given the known co-player.

Next Steps
----------

In future work, we aim to relax the pure memory-one assumption and
consider memory-two strategies or stochastic strategies. For stochastic
strategies, we'll need to address the issue of normalization in the posterior.

[1]: https://royalsocietypublishing.org/doi/full/10.1098/rspb.2021.1021

**Comment by yohm** [comment]
Thank you for a nice summary!

The question can also be formalized like the following (if I understand it correctly):
Suppose you know that your co-player is playing a pure memory-one strategy and that the co-player's strategy is equally randomly sampled from the memory-1 strategy space; what is your best strategy? The best strategy would be a combination of the "opening sequences" to infer the co-player's strategy and the subsequent memory-1 strategy that best responds against the inferred strategy.

I want to point out that the "opening sequences" are not necessarily fixed in advance. Instead, it can be conditioned on the co-player's initial moves.

Perhaps, the simplest setting would be to consider the error-free limit. Our goal is to identify the best strategy given the discount factor $\delta$ and the benefit of cooperation $b$. (Without loss of generality we can set $c=1$.)

**Comment by Nikoleta-v3** [comment]
> I want to point out that the "opening sequences" are not necessarily fixed in advance. Instead, it can be conditioned on the co-player's initial moves.

I agree and this would be ideal. The only issue is that if we condition on the co-player's actions then the number of sequences that we need to check increase by a lot.
